src/Shared/FileUtilities.cs

```diff
@@ -251,6 +247,70 @@ internal static String GetDirectoryNameOfFullPath(String fullPath)
             return null;
         }
 
+        /// <summary>
+        /// Gets the canonicalized full path of the provided path.
+        /// Path.GetFullPath The pre .Net 4.6.2 implementation of Path.GetFullPath is slow and creates strings in its work.
+        /// Therefore MSBuild has its own implementation on full framework.
+        /// Guidance for use: call this on all paths accepted through public entry
+        /// points that need normalization. After that point, only verify the path
+        /// is rooted, using ErrorUtilities.VerifyThrowPathRooted.
+        /// ASSUMES INPUT IS ALREADY UNESCAPED.
+        /// </summary>
+        internal static string NormalizePath(string path)
+        {
+            ErrorUtilities.VerifyThrowArgumentLength(path, "path");
+            string fullPath = GetFullPath(path);
+#if FEATURE_LEGACY_GETFULLPATH
+            if (NativeMethodsShared.IsWindows)
+            {
+                return fullPath;
+            }
+#endif
+            return FixFilePath(fullPath);
```

Comment:
Preserved the behavior of the original function here, not sure I understand why FixFilePath is specifically not called on net46 Windows, yet called on mono and netcore Windows